### ISSUE TITLE
[BUG] AWS EMR 6.12 NDS SF3k query9 Failure on g4dn.4xlarge

### ISSUE THREAD

**Issue by btong04** [NVIDIA/spark-rapids#9142]
Query9 failed on 3 consecutive runs on the same cluster with the following settings on EMR 6.12:
```
benchmark:
  name: amazon-emr-ec2-t4

  cluster:
    platform: emr

    availability-zone: us-west-2a
    version: 6.12.0
    bootstrap-action: s3://mahrens-test/aws-emr-bootstrap.sh # todo: fix hardcoding
    keypair: btong-spark-rapids

    driver:
      instance-type: m5dn.4xlarge

    workers:
      instance-type: g4dn.4xlarge
      num: 8

  scratch: mahrens-test # todo: should be s3://mahrens-test/scratch
  dataset: s3://ndsv2-data/parquet_sf3000
  dataset-format: parquet
  query-stream: query_streams/sf3k_query_0.sql
  queries: benchmark
  runs: 3
  config: |
    "--conf" "spark.driver.cores=8"
    "--conf" "spark.driver.maxResultSize=2gb"
    "--conf" "spark.driver.memory=30G"
    "--conf" "spark.dynamicAllocation.enabled=false"
    "--conf" "spark.eventLog.compress=true"
    "--conf" "spark.executor.cores=16"
    "--conf" "spark.executor.instances=8"
    "--conf" "spark.executor.memory=16G"
    "--conf" "spark.executor.memoryOverhead=16G"
    "--conf" "spark.executor.resource.gpu.amount=2"
    "--conf" "spark.locality.wait=0s"
    "--conf" "spark.rapids.filecache.enabled=true"
    "--conf" "spark.rapids.memory.host.spillStorageSize=32G"
    "--conf" "spark.rapids.memory.pinnedPool.size=8G"
    "--conf" "spark.rapids.shuffle.mode=MULTITHREADED"
    "--conf" "spark.rapids.sql.batchSizeBytes=1gb"
    "--conf" "spark.rapids.sql.concurrentGpuTasks=2"
    "--conf" "spark.rapids.sql.incompatibleOps.enabled=true"
    "--conf" "spark.shuffle.manager=com.nvidia.spark.rapids.spark340.RapidsShuffleManager"
    "--conf" "spark.sql.adaptive.enabled=true"
    "--conf" "spark.sql.files.maxPartitionBytes=2gb"
    "--conf" "spark.sql.sources.useV1SourceList=parquet"
    "--conf" "spark.task.resource.gpu.amount=0.00042"
    "--conf" "spark.yarn.unmanagedAM.enabled=true"
```

Error message:
```
"exceptions": [
    "An error occurred while calling o5060.collectToPython.\n: org.apache.spark.SparkException: Job aborted due to stage failure: Task 3 in stage 1038.0 failed 4 times, most recent failure: Lost task 3.3 in stage 1038.0 (TID 38970) (ip-172-31-54-86.us-west-2.compute.internal executor 7): com.nvidia.spark.rapids.jni.SplitAndRetryOOM: GPU OutOfMemory: could not split inputs and retry\n\tat com.nvidia.spark.rapids.RmmRapidsRetryIterator$AutoCloseableAttemptSpliterator.split(RmmRapidsRetryIterator.scala:410)\n\tat com.nvidia.spark.rapids.RmmRapidsRetryIterator$RmmRapidsRetryIterator.next(RmmRapidsRetryIterator.scala:520)\n\tat com.nvidia.spark.rapids.RmmRapidsRetryIterator$RmmRapidsRetryAutoCloseableIterator.next(RmmRapidsRetryIterator.scala:458)\n\tat com.nvidia.spark.rapids.RmmRapidsRetryIterator$.drainSingleWithVerification(RmmRapidsRetryIterator.scala:275)\n\tat com.nvidia.spark.rapids.RmmRapidsRetryIterator$.withRetryNoSplit(RmmRapidsRetryIterator.scala:128)\n\tat com.nvidia.spark.rapids.GpuTieredProject.projectAndCloseWithRetrySingleBatch(basicPhysicalOperators.scala:372)\n\tat com.nvidia.spark.rapids.GpuHashAggregateIterator$AggHelper.$anonfun$preProcess$1(aggregate.scala:271)\n\tat com.nvidia.spark.rapids.Arm$.withResource(Arm.scala:29)\n\tat com.nvidia.spark.rapids.GpuHashAggregateIterator$AggHelper.preProcess(aggregate.scala:270)\n\tat com.nvidia.spark.rapids.GpuHashAggregateIterator$.$anonfun$computeAggregateAndClose$1(aggregate.scala:421)\n\tat com.nvidia.spark.rapids.Arm$.withResource(Arm.scala:29)\n\tat com.nvidia.spark.rapids.GpuHashAggregateIterator$.computeAggregateAndClose(aggregate.scala:416)\n\tat com.nvidia.spark.rapids.GpuHashAggregateIterator.aggregateInputBatches(aggregate.scala:604)\n\tat com.nvidia.spark.rapids.GpuHashAggregateIterator.$anonfun$next$2(aggregate.scala:556)\n\tat scala.Option.getOrElse(Option.scala:189)\n\tat com.nvidia.spark.rapids.GpuHashAggregateIterator.next(aggregate.scala:553)\n\tat com.nvidia.spark.rapids.GpuHashAggregateIterator.next(aggregate.scala:498)\n\tat org.apache.spark.sql.rapids.execution.GpuShuffleExchangeExecBase$$anon$1.partNextBatch(GpuShuffleExchangeExecBase.scala:318)\n\tat org.apache.spark.sql.rapids.execution.GpuShuffleExchangeExecBase$$anon$1.hasNext(GpuShuffleExchangeExecBase.scala:340)\n\tat org.apache.spark.sql.rapids.RapidsShuffleThreadedWriterBase.$anonfun$write$2(RapidsShuffleInternalManagerBase.scala:281)\n\tat org.apache.spark.sql.rapids.RapidsShuffleThreadedWriterBase.$anonfun$write$2$adapted(RapidsShuffleInternalManagerBase.scala:274)\n\tat com.nvidia.spark.rapids.Arm$.withResource(Arm.scala:29)\n\tat org.apache.spark.sql.rapids.RapidsShuffleThreadedWriterBase.$anonfun$write$1(RapidsShuffleInternalManagerBase.scala:274)\n\tat org.apache.spark.sql.rapids.RapidsShuffleThreadedWriterBase.$anonfun$write$1$adapted(RapidsShuffleInternalManagerBase.scala:273)\n\tat com.nvidia.spark.rapids.Arm$.withResource(Arm.scala:29)\n\tat org.apache.spark.sql.rapids.RapidsShuffleThreadedWriterBase.write(RapidsShuffleInternalManagerBase.scala:273)\n\tat org.apache.spark.shuffle.ShuffleWriteProcessor.write(ShuffleWriteProcessor.scala:59)\n\tat org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:101)\n\tat org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:53)\n\tat org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:161)\n\tat org.apache.spark.scheduler.Task.run(Task.scala:141)\n\tat org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:554)\n\tat org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1541)\n\tat org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:557)\n\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)\n\tat java.lang.Thread.run(Thread.java:750)\n\nDriver stacktrace:\n\tat org.apache.spark.scheduler.DAGScheduler.failJobAndIndependentStages(DAGScheduler.scala:2974)\n\tat org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2(DAGScheduler.scala:2910)\n\tat org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2$adapted(DAGScheduler.scala:2909)\n\tat scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)\n\tat scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)\n\tat scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)\n\tat org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:2909)\n\tat org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1(DAGScheduler.scala:1263)\n\tat org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1$adapted(DAGScheduler.scala:1263)\n\tat scala.Option.foreach(Option.scala:407)\n\tat org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:1263)\n\tat org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:3173)\n\tat org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:3112)\n\tat org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:3101)\n\tat org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:49)\n\tat org.apache.spark.sql.execution.adaptive.AdaptiveExecutor.checkNoFailures(AdaptiveExecutor.scala:154)\n\tat org.apache.spark.sql.execution.adaptive.AdaptiveExecutor.doRun(AdaptiveExecutor.scala:88)\n\tat org.apache.spark.sql.execution.adaptive.AdaptiveExecutor.tryRunningAndGetFuture(AdaptiveExecutor.scala:66)\n\tat org.apache.spark.sql.execution.adaptive.AdaptiveExecutor.execute(AdaptiveExecutor.scala:57)\n\tat org.apache.spark.sql.execution.adaptive.AdaptiveSparkPlanExec.$anonfun$getFinalPhysicalPlan$1(AdaptiveSparkPlanExec.scala:265)\n\tat org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:827)\n\tat org.apache.spark.sql.execution.adaptive.AdaptiveSparkPlanExec.getFinalPhysicalPlan(AdaptiveSparkPlanExec.scala:264)\n\tat org.apache.spark.sql.execution.adaptive.AdaptiveSparkPlanExec.withFinalPlanUpdate(AdaptiveSparkPlanExec.scala:541)\n\tat org.apache.spark.sql.execution.adaptive.AdaptiveSparkPlanExec.executeCollect(AdaptiveSparkPlanExec.scala:503)\n\tat org.apache.spark.sql.Dataset.$anonfun$collectToPython$1(Dataset.scala:4068)\n\tat org.apache.spark.sql.Dataset.$anonfun$withAction$2(Dataset.scala:4234)\n\tat org.apache.spark.sql.execution.QueryExecution$.withInternalError(QueryExecution.scala:570)\n\tat org.apache.spark.sql.Dataset.$anonfun$withAction$1(Dataset.scala:4232)\n\tat org.apache.spark.sql.catalyst.QueryPlanningTracker$.withTracker(QueryPlanningTracker.scala:107)\n\tat org.apache.spark.sql.execution.SQLExecution$.withTracker(SQLExecution.scala:250)\n\tat org.apache.spark.sql.execution.SQLExecution$.executeQuery$1(SQLExecution.scala:123)\n\tat org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$9(SQLExecution.scala:160)\n\tat org.apache.spark.sql.catalyst.QueryPlanningTracker$.withTracker(QueryPlanningTracker.scala:107)\n\tat org.apache.spark.sql.execution.SQLExecution$.withTracker(SQLExecution.scala:250)\n\tat org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$8(SQLExecution.scala:160)\n\tat org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:271)\n\tat org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$1(SQLExecution.scala:159)\n\tat org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:827)\n\tat org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:69)\n\tat org.apache.spark.sql.Dataset.withAction(Dataset.scala:4232)\n\tat org.apache.spark.sql.Dataset.collectToPython(Dataset.scala:4065)\n\tat sun.reflect.GeneratedMethodAccessor118.invoke(Unknown Source)\n\tat sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)\n\tat java.lang.reflect.Method.invoke(Method.java:498)\n\tat py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)\n\tat py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)\n\tat py4j.Gateway.invoke(Gateway.java:282)\n\tat py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)\n\tat py4j.commands.CallCommand.execute(CallCommand.java:79)\n\tat py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)\n\tat py4j.ClientServerConnection.run(ClientServerConnection.java:106)\n\tat java.lang.Thread.run(Thread.java:750)\nCaused by: com.nvidia.spark.rapids.jni.SplitAndRetryOOM: GPU OutOfMemory: could not split inputs and retry\n\tat com.nvidia.spark.rapids.RmmRapidsRetryIterator$AutoCloseableAttemptSpliterator.split(RmmRapidsRetryIterator.scala:410)\n\tat com.nvidia.spark.rapids.RmmRapidsRetryIterator$RmmRapidsRetryIterator.next(RmmRapidsRetryIterator.scala:520)\n\tat com.nvidia.spark.rapids.RmmRapidsRetryIterator$RmmRapidsRetryAutoCloseableIterator.next(RmmRapidsRetryIterator.scala:458)\n\tat com.nvidia.spark.rapids.RmmRapidsRetryIterator$.drainSingleWithVerification(RmmRapidsRetryIterator.scala:275)\n\tat com.nvidia.spark.rapids.RmmRapidsRetryIterator$.withRetryNoSplit(RmmRapidsRetryIterator.scala:128)\n\tat com.nvidia.spark.rapids.GpuTieredProject.projectAndCloseWithRetrySingleBatch(basicPhysicalOperators.scala:372)\n\tat com.nvidia.spark.rapids.GpuHashAggregateIterator$AggHelper.$anonfun$preProcess$1(aggregate.scala:271)\n\tat com.nvidia.spark.rapids.Arm$.withResource(Arm.scala:29)\n\tat com.nvidia.spark.rapids.GpuHashAggregateIterator$AggHelper.preProcess(aggregate.scala:270)\n\tat com.nvidia.spark.rapids.GpuHashAggregateIterator$.$anonfun$computeAggregateAndClose$1(aggregate.scala:421)\n\tat com.nvidia.spark.rapids.Arm$.withResource(Arm.scala:29)\n\tat com.nvidia.spark.rapids.GpuHashAggregateIterator$.computeAggregateAndClose(aggregate.scala:416)\n\tat com.nvidia.spark.rapids.GpuHashAggregateIterator.aggregateInputBatches(aggregate.scala:604)\n\tat com.nvidia.spark.rapids.GpuHashAggregateIterator.$anonfun$next$2(aggregate.scala:556)\n\tat scala.Option.getOrElse(Option.scala:189)\n\tat com.nvidia.spark.rapids.GpuHashAggregateIterator.next(aggregate.scala:553)\n\tat com.nvidia.spark.rapids.GpuHashAggregateIterator.next(aggregate.scala:498)\n\tat org.apache.spark.sql.rapids.execution.GpuShuffleExchangeExecBase$$anon$1.partNextBatch(GpuShuffleExchangeExecBase.scala:318)\n\tat org.apache.spark.sql.rapids.execution.GpuShuffleExchangeExecBase$$anon$1.hasNext(GpuShuffleExchangeExecBase.scala:340)\n\tat org.apache.spark.sql.rapids.RapidsShuffleThreadedWriterBase.$anonfun$write$2(RapidsShuffleInternalManagerBase.scala:281)\n\tat org.apache.spark.sql.rapids.RapidsShuffleThreadedWriterBase.$anonfun$write$2$adapted(RapidsShuffleInternalManagerBase.scala:274)\n\tat com.nvidia.spark.rapids.Arm$.withResource(Arm.scala:29)\n\tat org.apache.spark.sql.rapids.RapidsShuffleThreadedWriterBase.$anonfun$write$1(RapidsShuffleInternalManagerBase.scala:274)\n\tat org.apache.spark.sql.rapids.RapidsShuffleThreadedWriterBase.$anonfun$write$1$adapted(RapidsShuffleInternalManagerBase.scala:273)\n\tat com.nvidia.spark.rapids.Arm$.withResource(Arm.scala:29)\n\tat org.apache.spark.sql.rapids.RapidsShuffleThreadedWriterBase.write(RapidsShuffleInternalManagerBase.scala:273)\n\tat org.apache.spark.shuffle.ShuffleWriteProcessor.write(ShuffleWriteProcessor.scala:59)\n\tat org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:101)\n\tat org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:53)\n\tat org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:161)\n\tat org.apache.spark.scheduler.Task.run(Task.scala:141)\n\tat org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:554)\n\tat org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1541)\n\tat org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:557)\n\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)\n\t... 1 more\n"
```

Also tried again with `spark.executor.resource.gpu.amount=1` and got same `com.nvidia.spark.rapids.jni.SplitAndRetryOOM: GPU OutOfMemory: could not split inputs and retry` error.

**Comment by mattahrens** [comment]
I reproduced this query 9 issue with EMR 6.12.0 on all 6 of my runs this morning: 
```
ERROR BEGIN
An error occurred while calling o247.collectToPython.
: org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 31.0 failed 4 times, most recent failure: Lost task 0.3 in stage 31.0 (TID 12267) (ip-172-31-48-216.us-west-2.compute.internal executor 3): com.nvidia.spark.rapids.jni.SplitAndRetryOOM: GPU OutOfMemory: could not split inputs and retry
        at com.nvidia.spark.rapids.RmmRapidsRetryIterator$AutoCloseableAttemptSpliterator.split(RmmRapidsRetryIterator.scala:410)
        at com.nvidia.spark.rapids.RmmRapidsRetryIterator$RmmRapidsRetryIterator.next(RmmRapidsRetryIterator.scala:520)
        at com.nvidia.spark.rapids.RmmRapidsRetryIterator$RmmRapidsRetryAutoCloseableIterator.next(RmmRapidsRetryIterator.scala:458)
        at com.nvidia.spark.rapids.RmmRapidsRetryIterator$.drainSingleWithVerification(RmmRapidsRetryIterator.scala:275)
        at com.nvidia.spark.rapids.RmmRapidsRetryIterator$.withRetryNoSplit(RmmRapidsRetryIterator.scala:128)
        at com.nvidia.spark.rapids.GpuTieredProject.projectAndCloseWithRetrySingleBatch(basicPhysicalOperators.scala:372)
        at com.nvidia.spark.rapids.GpuHashAggregateIterator$AggHelper.$anonfun$preProcess$1(aggregate.scala:271)
        at com.nvidia.spark.rapids.Arm$.withResource(Arm.scala:29)
        at com.nvidia.spark.rapids.GpuHashAggregateIterator$AggHelper.preProcess(aggregate.scala:270)
        at com.nvidia.spark.rapids.GpuHashAggregateIterator$.$anonfun$computeAggregateAndClose$1(aggregate.scala:421)
        at com.nvidia.spark.rapids.Arm$.withResource(Arm.scala:29)
        at com.nvidia.spark.rapids.GpuHashAggregateIterator$.computeAggregateAndClose(aggregate.scala:416)
        at com.nvidia.spark.rapids.GpuHashAggregateIterator.aggregateInputBatches(aggregate.scala:604)
        at com.nvidia.spark.rapids.GpuHashAggregateIterator.$anonfun$next$2(aggregate.scala:556)
        at scala.Option.getOrElse(Option.scala:189)
        at com.nvidia.spark.rapids.GpuHashAggregateIterator.next(aggregate.scala:553)
        at com.nvidia.spark.rapids.GpuHashAggregateIterator.next(aggregate.scala:498)
        at org.apache.spark.sql.rapids.execution.GpuShuffleExchangeExecBase$$anon$1.partNextBatch(GpuShuffleExchangeExecBase.scala:318)
        at org.apache.spark.sql.rapids.execution.GpuShuffleExchangeExecBase$$anon$1.hasNext(GpuShuffleExchangeExecBase.scala:340)
        at org.apache.spark.sql.rapids.RapidsShuffleThreadedWriterBase.$anonfun$write$2(RapidsShuffleInternalManagerBase.scala:281)
        at org.apache.spark.sql.rapids.RapidsShuffleThreadedWriterBase.$anonfun$write$2$adapted(RapidsShuffleInternalManagerBase.scala:274)
        at com.nvidia.spark.rapids.Arm$.withResource(Arm.scala:29)
        at org.apache.spark.sql.rapids.RapidsShuffleThreadedWriterBase.$anonfun$write$1(RapidsShuffleInternalManagerBase.scala:274)
        at org.apache.spark.sql.rapids.RapidsShuffleThreadedWriterBase.$anonfun$write$1$adapted(RapidsShuffleInternalManagerBase.scala:273)
        at com.nvidia.spark.rapids.Arm$.withResource(Arm.scala:29)
        at org.apache.spark.sql.rapids.RapidsShuffleThreadedWriterBase.write(RapidsShuffleInternalManagerBase.scala:273)
        at org.apache.spark.shuffle.ShuffleWriteProcessor.write(ShuffleWriteProcessor.scala:59)
        at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:101)
        at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:53)
        at org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:161)
        at org.apache.spark.scheduler.Task.run(Task.scala:141)
        at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:554)
        at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1541)
        at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:557)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:750)
```


Config used:
```
export SPARK_CONF=("--master" "yarn"
                   "--deploy-mode" "client"
                   "--conf" "spark.driver.maxResultSize=2GB"
                   "--conf" "spark.driver.memory=50G"
                   "--conf" "spark.executor.cores=12"
                   "--conf" "spark.executor.instances=16"
                   "--conf" "spark.executor.resource.gpu.amount=1"
                   "--conf" "spark.rapids.sql.concurrentGpuTasks=2"
                   "--conf" "spark.task.resource.gpu.amount=0.0833"
                   "--conf" "spark.executor.memory=16G"
                   "--conf" "spark.sql.files.maxPartitionBytes=2gb"
                   "--conf" "spark.rapids.memory.host.spillStorageSize=32G"
                   "--conf" "spark.rapids.memory.pinnedPool.size=8g"
                   "--conf" "spark.executor.memoryOverhead=16G"
                   "--conf" "spark.scheduler.minRegisteredResourcesRatio=1.0"
                   "--conf" "spark.sql.adaptive.enabled=true"
                   "--conf" "spark.driver.extraClassPath=$SPARK_RAPIDS_PLUGIN_JAR"
                   "--conf" "spark.executor.extraClassPath=$SPARK_RAPIDS_PLUGIN_JAR"
                   "--conf" "spark.shuffle.manager=com.nvidia.spark.rapids.spark340.RapidsShuffleManager"
                   "--conf" "spark.rapids.shuffle.multiThreaded.writer.threads=16"
                   "--conf" "spark.rapids.shuffle.multiThreaded.reader.threads=16"
                   "--conf" "spark.rapids.shuffle.mode=MULTITHREADED"
                   "--jars" "$SPARK_RAPIDS_PLUGIN_JAR,$NDS_LISTENER_JAR")
```

**Comment by abellina** [comment]
Did any gpu objects complain about being leaked? I assume this used to work and now doesn't.

**Comment by jlowe** [comment]
Note that I have not seen this when running with 23.10 on EMR 6.12 (patching the jars with a custom build).  This OOM is happening at the preprocess of an aggregate and thus might have been fixed by #8618.

**Comment by mattahrens** [comment]
Cannot reproduce issue with latest patch jar for 23.10, so this should be resolved now.

**Comment by mattf** [comment]
@btong04 is this resolved for you?

**Comment by btong04** [comment]
Yes, this issue appears to be resolved by using the patched `rapids-4-spark_2.12-23.10.0-SNAPSHOT-cuda11-jlowe-shufflefix2.jar`.

**Comment by NVnavkumar** [comment]
Confirming that this also happens on EMR 6.13 (which also uses 23.06). Should confirm this is resolved on EMR 6.14
